src/sqlacodegen/generators.py

```diff
@@ -460,7 +459,7 @@ def render_column(
             kwargs["key"] = column.key
         if is_primary:
             kwargs["primary_key"] = True
-        if not column.nullable and not is_sole_pk and is_table:
+        if not column.nullable and not column.primary_key:
```

Comment:
It is possible for *some* columns of a multi-column primary key constraint to be `NULL`, though this is exceedingly rare in practice. But this is why I had the condition in place. Perhaps we could simply drop the `is_table` condition?

Reply:
Hmmmm. I do understand this concern. However, this resulted in a somewhat weird behavior where some composite primary keys columns were getting the nullability flag. Maybe I misunderstood what I was seeing. I'll return this flag and check again

Reply:
I'm not sure we have a test covering the case of partially nullable PK, but it might be worth adding one if we don't have it already.

Reply:
Ok, I realized what was bugging. I've improved the conditioning and added a test to cover exactly the case you mentioned